docs/src/utils/getPagesManifest.ts

```diff
@@ -0,0 +1,21 @@
+function pluckMeta({ frontmatter, href, slug }) {
+  return { frontmatter, href, slug };
+}
+export async function getPagesManifest(
+  getContentPaths,
+  getPageFromSlug,
+  default_meta_info
+) {
+  const paths = await getContentPaths();
+  const pages = (await Promise.all(
+    paths.map(getPageFromSlug).map((page) => page.then(pluckMeta))
+  )) as any[];
+
+  return {
+    ...pages.reduce(
```

Comment:
Reduce is hard to read. Can you change this to a simple loop instead?

Reply:
I personally think `for...loop` doesn't look as clean as `reduce` and it's not easier to understand in this case, is it?